api/search/pathparser.py

```diff
@@ -0,0 +1,98 @@
+"""
+The purpose of this module is to parse a query path
+(expected in a xpath-like format).
+we use only "/" and "//" to indicate respectively children and generic descendants.
+
+Examples:
+"collections/sessions/acquisitions/files" and "collections//acquisitions/files"
+will prepare a search for files in an acquisition in a collection
+
+"collections//files"
+a search for files in collections, sessions and acquisitions (belonging to a collections)
+
+"acquisitions" and "projects//acquisitions"
+a search for acquisitions (as incidentally every acquisition is included in a project)
+
+"""
+from collections import deque as Queue
+
+class PathNotValidException(Exception):
+    pass
+
+
+class PathParser:
+
+    _search_graph = {
+        'collections': ['sessions', 'notes', 'files'],
+        'groups': ['projects'],
+        'projects': ['sessions', 'notes', 'files'],
+        'sessions': ['acquisitions', 'notes', 'files'],
+        'acquisitions': ['notes', 'files']
+    }
+
+    def __init__(self, path):
+        self.path = path
+        self.nodes = path.split('/')
+        if not (self.nodes[0]):
```

Comment:
You'll want to switch this to a `len(nodes) > 0`, otherwise you'll get an `IndexError` rather than the `False` you're expecting.